src/handleAction.js

```diff
@@ -7,19 +7,15 @@ export default function handleAction(type, reducers, defaultState) {
     ? type.toString()
     : type;
 
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw];
+
   return (state = defaultState, action) => {
     // If action type does not match, return previous state
     if (action.type !== typeValue) return state;
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
-
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
-    }
-
-    // Otherwise, assume an action map was passed
-    const reducer = reducers[handlerKey];
+    const reducer = action.error === true ? throwReducer : nextReducer;
 
     return isFunction(reducer)
       ? reducer(state, action)
```

Comment:
> Note that these checks would make a use case of factoring out...

That reminds me– are we trying to keep this project minimally dependent on others? Because lodash has good implementations of both (isFunction)[https://lodash.com/docs#isFunction] and (ownKeys)[https://lodash.com/docs#keys] we could use.